mussel

```diff
@@ -559,6 +561,33 @@ while [ $# -gt 0 ]; do
   g | -g | --disable-isl)
     ISL_SUPPORT=no
     ;;
+  T | -T | --custom-vendor)
+    # Have the user defined the default tuple first?
+    if [ -z $XTARGET ]; then
+      printf '%b!!%b '\''%s'\'' requires that the target architecture is already specified.\n' \
+        "$REDC" "$NORMALC" $1
+      printf 'Run '\''$0 -h'\'' for help.\n' $0
+      exit 1
+    fi
+    # Set the vendor as the next argument to this flag.
+    shift
+    CUSTOM_VENDOR="$1"
+    if (echo "$CUSTOM_VENDOR" | grep ' ') 2>&1 >/dev/null; then
+      printf '%b!!%b Illegal vendor '\''%s'\'': spaces are not permitted.\n' \
+        "$REDC" "$NORMALC" "$VENDOR"
+      exit 1
+    fi
+    OLD_XTARGET=$XTARGET
+    # Yes, we're placing the vendor before the target.
+    # Trust me, it will make sense in the AWK script.
+    XTARGET="$(printf '%s-%s' $CUSTOM_VENDOR $OLD_XTARGET | awk '{
```

Comment:
And perhaps I could do this using that POSIX "regex" at the parameter expansions instead of using awk, even though it isn't overkill here.

Reply:
Again I don't think that will be necessary as well, as awk is expected to be available on host systems where mussel will run.

Reply:
Already made it. Not like speed would matter since it's a single string, but it's, besides faster, simpler.